dali/pipeline/basic/coords.h

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_BASIC_COORDS_H_
+#define DALI_PIPELINE_BASIC_COORDS_H_
+
+#include "dali/common.h"
+#include "dali/pipeline/data/types.h"
+
+namespace dali {
+namespace basic {
+
+template <typename T>
+Index getPlane(const T& shape) {
+  return 1;
+}
+
+template <size_t I, size_t N, typename T>
+Index getPlane(const T& shape) {
```

Comment:
For me it's not very clear, what `I`, `N` and `T` are, what is the return value also. Maybe this function could have some kind of documentation?
Unless it is in fact clear, it's just me who sucks at templates ;)

Reply:
Done